paymentsheet/src/main/java/com/stripe/android/paymentsheet/navigation/BackStackHandling.kt

```diff
@@ -0,0 +1,76 @@
+@file:Suppress("MatchingDeclarationName")
+
+package com.stripe.android.paymentsheet.navigation
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import com.stripe.android.paymentsheet.BaseAddPaymentMethodFragment
+import com.stripe.android.paymentsheet.BasePaymentMethodsListFragment
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.TransitionTarget
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.TransitionTarget.AddAnotherPaymentMethod
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.TransitionTarget.AddFirstPaymentMethod
+import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel.TransitionTarget.SelectSavedPaymentMethods
+
+internal sealed interface NavigationEffect {
```

Comment:
This is a workaround for the time being. Eventually, we’ll observe `viewModel.currentScreen` in Compose. Until then, we need to map from the state-based world to the event-based world.